AirCasting/CoreData/CoreDataHook.swift

```diff
@@ -0,0 +1,54 @@
+// Created by Lunar on 17/05/2021.
+//
+
+import Foundation
+import CoreData
+
+final class CoreDataHook: NSObject, ObservableObject {
+    
+    @Published var sessions: [Session] = []
+    var context: NSManagedObjectContext!
+    private var fetchedResultsController: NSFetchedResultsController<Session>!
+    
+    func setup(selectedSection: SelectedSection) throws {
+        
+        let request = NSFetchRequest<Session>(entityName: "Session")
+        
+        switch selectedSection {
+        case .fixed:
+            request.predicate = NSPredicate(format: "type == %@", SessionType.fixed.rawValue)
+        case .mobileActive:
+            request.predicate = NSPredicate(format: "type == %@ AND status == %li", SessionType.mobile.rawValue, SessionStatus.RECORDING.rawValue)
+        case .mobileDormant:
+            request.predicate = NSPredicate(format: "type == %@ AND status == %li", SessionType.mobile.rawValue, SessionStatus.FINISHED.rawValue)
+        case .following:
+            request.predicate = NSPredicate(format: "followedAt != NULL")
+        }
+        
+        request.sortDescriptors = [NSSortDescriptor(key: "startTime", ascending: true)]
```

Comment:
this can be set only once just before NSFetchedResultsController init so You can just capture `let predicate`